engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/parser/EventSubscriptionDeclaration.java

```diff
@@ -54,6 +55,8 @@ public class EventSubscriptionDeclaration implements Serializable {
 
   protected EventSubscriptionJobDeclaration jobDeclaration = null;
 
+  BaseDelegateExecution contextExecution;
```

Comment:
Storing the execution in a field is not thread-safe, because we have only one declaration per process definition. Please try to find a solution that hands in the execution directly during the invocation of the expression. Maybe it is possible to reuse the `VariableScope` argument of `#resolveExpressionOfEventName`.